src/pages/HomeParticipant.vue

```diff
@@ -232,38 +237,35 @@ async function checkConsent() {
 
   consentType.value = docType;
 
-  const consentStatus = _get(userData.value, `legal.${consentType.value}`);
+  const consentStatus = userData.value?.legal?.[consentType.value];
   const consentDoc = await authStore.getLegalDoc(docType);
   consentVersion.value = consentDoc.version;
 
-  if (_get(toRaw(consentStatus), consentDoc.version)) {
-    const legalDocs = _get(toRaw(consentStatus), consentDoc.version);
+  if (consentStatus?.[consentDoc.version]) {
+    const legalDocs = consentStatus?.[consentDoc.version];
+
     let found = false;
     let signedBeforeAugFirst = false;
-    let signedAfterAugFirst = false;
 
-    _forEach(legalDocs, (document) => {
+    for (const document of legalDocs) {
```

Comment:
I just want to understand: Why not using _forEach to optimize for performance by stopping the loop early?

Reply:
I made the change because, just from experience, `lodash` only comes with benefits in specific cases  – the rest of the time it's an unnecessary dependency with (modern) native JS utilities being able to do the exact same. There are however cases where, whilst doable in native JS, using a `lodash` method is cleaner / improves readability. 

In this case though, for such as simple loop, lodash is a bit overkill in my opinion. You are however completely right, since this loop is actually just intended to find a single element, braking the loop as soon as we have found a result is beneficial. I had overlooked this since until now the breaker was only applied if we found a document signed after the limit date. 

Anyway, here's a little JSPerf example to show the performance differences on a slightly simplified example that runs on a collection of 10 documents: https://jsperf.app/rakoqo/3 

![jsperf app_rakoqo_3](https://github.com/user-attachments/assets/e3233679-1581-41b6-8c47-0c5968889498)

Reply:
This explanation is so informative, thank you for sharing :)